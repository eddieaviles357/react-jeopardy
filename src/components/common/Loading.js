import React, { useRef } from 'react'
import { useAnimationFrame } from 'framer-motion'

const Loading = () => {
  const ref = useRef()

  useAnimationFrame((time, delta) => {
    if (!ref.current) return;
    ref.current.style.transform = `rotate(${time / 10}deg`
  })

  return (
    <div
      ref={ref}
      alt='Loading'
      className='Loading'>
      {/*Used only for testing purpose*/}
      <div style={{ display: 'none' }}>Loading</div>
    </div>
  )
}

export default Loading
