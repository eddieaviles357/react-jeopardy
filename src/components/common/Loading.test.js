"use strict";

import { render } from '@testing-library/react'
import Loading from './Loading';

test('does not crash', () => {
    render(<Loading />)
})

test('matches snapshot', () => {
    const loading = render(<Loading />);
    expect(loading.asFragment()).toMatchSnapshot();
})