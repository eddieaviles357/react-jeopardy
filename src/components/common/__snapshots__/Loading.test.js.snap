// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    alt="Loading"
    class="Loading"
  >
    <div
      style="display: none;"
    >
      Loading
    </div>
  </div>
</DocumentFragment>
`;
